src/main/scala/com/rbmhtechnology/calliope/scaladsl/TransactionalEventProducer.scala

```diff
@@ -203,68 +253,140 @@ private class GraphRunnerSupervisor(graph: ProducerGraph, onFailure: FailureHand
 private object ProducerGraphRunner {
 
   case object NotifyCommit
+  case object RunGraph
+  case object StopGraph
+
+  case object GraphStarted
+  case object GraphStopped
 
-  private[ProducerGraphRunner] case object StartGraph
   private[ProducerGraphRunner] case class RestartGraph(failure: Throwable)
 
+  private[ProducerGraphRunner] case class GraphAbortedException() extends RuntimeException
+
+  private[ProducerGraphRunner] sealed trait State
+  private[ProducerGraphRunner] case object Stopped extends State
+  private[ProducerGraphRunner] case object Stopping extends State
+  private[ProducerGraphRunner] case object Running extends State
+
+  private[ProducerGraphRunner] case object Data {
+    def apply(): Data =
+      Data(None, None, Vector.empty)
```

Comment:
Why aren't those default values of the parameters (so we could omit the object definition)?